code/items/equipments/Hands.cs

```diff
@@ -0,0 +1,191 @@
+
+        public override void Simulate(Client client)
+        {
+            base.Simulate(client);
+
+            if (!IsServer)
+            {
+                return;
+            }
+
+            if (Owner is not TTTPlayer player)
+            {
+                return;
+            }
+
+            using (Prediction.Off())
+            {
+                if (Input.Pressed(InputButton.Attack1))
+                {
+                    TryGrabEntity(player);
+                }
+                else if (Input.Pressed(InputButton.Attack2))
+                {
+                    if (IsHoldingEntity)
+                    {
+                        GrabbedEntity?.Drop();
+                    }
+                    else
+                    {
+                        PushPlayer(player);
+                    }
+                }
+                else if (Input.Pressed(InputButton.Reload) && IsHoldingEntity)
+                {
+                    GrabbedEntity?.SecondaryAction();
+                }
+
+                GrabbedEntity?.Update(player);
+            }
+        }
+
+        private void PushPlayer(TTTPlayer player)
+        {
+            TraceResult tr = Trace.Ray(player.EyePos, player.EyePos + player.EyeRot.Forward * MAX_INTERACT_DISTANCE)
+                    .Ignore(player)
+                    .Run();
+
+            if (!tr.Hit || tr.Entity is not TTTPlayer || !tr.Entity.IsValid())
+            {
+                return;
+            }
+
+            IsPushingPlayer = true;
+
+            player.SetAnimBool("b_attack", true);
+            player.SetAnimInt("holdtype", 4);
+            player.SetAnimInt("holdtype_handedness", 0);
+
+            tr.Entity.Velocity += player.EyeRot.Forward * PUSHING_FORCE;
+
+            _ = WaitForAnimationFinish();
+        }
+
+        private async Task WaitForAnimationFinish()
+        {
+            await GameTask.DelaySeconds(0.6f);
+            IsPushingPlayer = false;
+        }
+
+        private void TryGrabEntity(TTTPlayer player)
+        {
+            if (IsHoldingEntity)
+            {
+                return;
+            }
+
+            Vector3 eyePos = player.EyePos;
+            Vector3 eyeDir = player.EyeRot.Forward;
+
+            TraceResult tr = Trace.Ray(eyePos, eyePos + eyeDir * MAX_INTERACT_DISTANCE)
+                .UseHitboxes()
+                .Ignore(player)
+                .HitLayer(CollisionLayer.Debris)
+                .EntitiesOnly()
+                .Run();
+
+            // Make sure trace is hit and not null.
+            if (!tr.Hit || !tr.Entity.IsValid())
+            {
+                return;
+            }
+
+            // Cannot pickup items held by other players.
+            if (tr.Entity.Parent != null)
+            {
+                return;
+            }
+
+            switch (tr.Entity)
+            {
+                case PlayerCorpse:
+                    GrabbedEntity = new GrabbableCorpse(player, tr.Entity as PlayerCorpse, tr.Body, tr.Bone);
```

Comment:
You're switching over `tr.Entity`, here checking for `PlayerCorpse`, but then casting it into a `PlayerCorpse`?